File: composeApp/src/commonMain/kotlin/com/beekeeper/app/domain/model/Inspection.kt
package com.beekeeper.app.domain.model

import kotlinx.datetime.LocalDateTime
import kotlinx.serialization.Serializable

@Serializable
data class Inspection(
    val id: String,
    val hiveId: String,
    val userId: String,
    val inspectionDate: LocalDateTime,
    val durationMinutes: Int? = null,
    val queenSeen: Boolean = false,
    val queenMarked: Boolean = false,
    val queenColor: String? = null,
    val queenCells: QueenCellStatus = QueenCellStatus.NONE,
    val eggsSeen: Boolean = false,
    val larvaeSeen: Boolean = false,
    val cappedBroodSeen: Boolean = false,
    val broodPattern: BroodPattern = BroodPattern.GOOD,
    val broodFrames: Int? = null,
    val population: ColonyPopulation = ColonyPopulation.MEDIUM,
    val temperament: ColonyTemperament = ColonyTemperament.CALM,
    val healthStatus: InspectionHealthStatus = InspectionHealthStatus.HEALTHY,
    val honeyStores: ResourceLevel = ResourceLevel.MEDIUM,
    val honeyFrames: Int? = null,
    val pollenStores: ResourceLevel = ResourceLevel.MEDIUM,
    val pollenFrames: Int? = null,
    val varroaMitesDetected: Boolean = false,
    val varroaLevel: String? = null,
    val otherPestsDetected: Boolean = false,
    val pestDescription: String? = null,
    val diseaseDetected: Boolean = false,
    val diseaseDescription: String? = null,
    val feedingDone: Boolean = false,
    val feedingType: String? = null,
    val feedingAmount: String? = null,
    val treatmentApplied: Boolean = false,
    val treatmentType: String? = null,
    val equipmentChanges: String? = null,
    val weatherTemp: Double? = null,
    val weatherConditions: String? = null,
    val notes: String? = null,
    val photos: List<String>? = null,
    val nextInspectionDate: LocalDateTime? = null,
    val createdAt: LocalDateTime,
    val updatedAt: LocalDateTime
)

enum class QueenCellStatus {
    NONE,
    PLAY_CUPS,
    QUEEN_CELLS,
    CHARGED_CELLS,
    CAPPED_CELLS,
    EMERGENCY_CELLS
}

enum class BroodPattern {
    EXCELLENT,
    GOOD,
    FAIR,
    POOR,
    SPOTTY
}

enum class ColonyTemperament {
    VERY_CALM,
    CALM,
    MODERATE,
    DEFENSIVE,
    VERY_DEFENSIVE,
    AGGRESSIVE
}

enum class ColonyPopulation {
    VERY_STRONG,
    STRONG,
    MEDIUM,
    WEAK,
    VERY_WEAK
}

enum class InspectionHealthStatus {
    EXCELLENT,
    HEALTHY,
    FAIR,
    CONCERNING,
    CRITICAL
}

enum class ResourceLevel {
    FULL,
    HIGH,
    MEDIUM,
    LOW,
    EMPTY
}

File: api/app/schemas/inspection.py
from pydantic import BaseModel, ConfigDict
from datetime import datetime
from typing import Optional
from app.models import (
    QueenCellStatus,
    BroodPattern,
    ColonyTemperament,
    ColonyPopulation,
    HealthStatus,
    ResourceLevel,
)


def to_camel(string: str) -> str:
    """Convert snake_case to camelCase"""
    words = string.split('_')
    return words[0] + ''.join(word.capitalize() for word in words[1:])


class InspectionBase(BaseModel):
    hive_id: str
    inspection_date: datetime
    duration_minutes: Optional[int] = None

    # Weather
    weather_temp: Optional[float] = None
    weather_conditions: Optional[str] = None

    # Queen observations
    queen_seen: bool = False
    queen_marked: bool = False
    queen_cells: QueenCellStatus = QueenCellStatus.NONE

    # Brood observations
    brood_pattern: BroodPattern = BroodPattern.GOOD
    eggs_seen: bool = False
    larvae_seen: bool = False
    capped_brood_seen: bool = False
    estimated_brood_frames: Optional[int] = None

    # Colony observations
    temperament: ColonyTemperament = ColonyTemperament.CALM
    population: ColonyPopulation = ColonyPopulation.MEDIUM
    estimated_frames_covered: Optional[int] = None

    # Health observations
    health_status: HealthStatus = HealthStatus.HEALTHY
    varroa_mites_detected: bool = False
    pests_notes: str = ""
    disease_signs: str = ""

    # Resources
    honey_stores: ResourceLevel = ResourceLevel.ADEQUATE
    pollen_stores: ResourceLevel = ResourceLevel.ADEQUATE
    capped_honey: bool = False

    # Actions taken
    actions_taken: str = ""
    feeding_done: bool = False
    feeding_notes: str = ""
    treatment_applied: bool = False
    treatment_notes: str = ""

    # Media and notes
    photos: str = ""  # JSON array string or comma-separated URLs
    notes: str = ""
    next_inspection_date: Optional[datetime] = None

    model_config = ConfigDict(alias_generator=to_camel, populate_by_name=True)


class InspectionCreate(InspectionBase):
    """Schema for creating a new inspection"""
    pass


class InspectionUpdate(BaseModel):
    """Schema for updating an inspection"""
    inspection_date: Optional[datetime] = None
    duration_minutes: Optional[int] = None
    weather_temp: Optional[float] = None
    weather_conditions: Optional[str] = None
    queen_seen: Optional[bool] = None
    queen_marked: Optional[bool] = None
    queen_cells: Optional[QueenCellStatus] = None
    brood_pattern: Optional[BroodPattern] = None
    eggs_seen: Optional[bool] = None
    larvae_seen: Optional[bool] = None
    capped_brood_seen: Optional[bool] = None
    estimated_brood_frames: Optional[int] = None
    temperament: Optional[ColonyTemperament] = None
    population: Optional[ColonyPopulation] = None
    estimated_frames_covered: Optional[int] = None
    health_status: Optional[HealthStatus] = None
    varroa_mites_detected: Optional[bool] = None
    pests_notes: Optional[str] = None
    disease_signs: Optional[str] = None
    honey_stores: Optional[ResourceLevel] = None
    pollen_stores: Optional[ResourceLevel] = None
    capped_honey: Optional[bool] = None
    actions_taken: Optional[str] = None
    feeding_done: Optional[bool] = None
    feeding_notes: Optional[str] = None
    treatment_applied: Optional[bool] = None
    treatment_notes: Optional[str] = None
    photos: Optional[str] = None
    notes: Optional[str] = None
    next_inspection_date: Optional[datetime] = None

    model_config = ConfigDict(alias_generator=to_camel, populate_by_name=True)


class InspectionResponse(InspectionBase):
    """Schema for inspection responses"""
    id: str
    user_id: str
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True, alias_generator=to_camel, populate_by_name=True)
